packages/ra-auth-msal/src/authProvider.ts

```diff
@@ -153,6 +159,13 @@ export const msalAuthProvider = ({
         throw new Error("Authentication failed");
       }
       msalInstance.setActiveAccount(account);
+
+      // We cannot use react-router redirect here, as we are not in a router context,
+      // So we need to fallback to native browser APIs.
+      const redirectUrl =
+        sessionStorage.getItem(MSAL_REDIRECT_KEY) ?? window.location.origin;
+      window.location.replace(redirectUrl);
+      sessionStorage.removeItem(MSAL_REDIRECT_KEY);
```

Comment:
```suggestion
      // We cannot use react-router redirect here, as we are not in a router context,
      // So we need to fallback to native browser APIs.
      const redirectUrl =
        sessionStorage.getItem(MSAL_REDIRECT_KEY);
      if (redirectUrl) {
        window.location.replace(redirectUrl);
        sessionStorage.removeItem(MSAL_REDIRECT_KEY);
      }
```

Besides, I would prefer if there was a way do diable this feature, in case `window.location.href` creates issues (you can add an option to disable it).